druid-handler/src/java/org/apache/hadoop/hive/ql/io/DruidOutputFormat.java

```diff
@@ -0,0 +1,393 @@
+                try {
+                  log.info("did pushed the segment [%s]", pushedSegment);
+                  // maybe write to disk the segment location to publish ?
+                }
+                catch (Exception e) {
+                  Throwables.propagate(e);
+                }
+              }
+
+              log.info("Published [%,d] sinks.", segmentsToPush.size());
+            }
+
+            @Override
+            public void onFailure(Throwable e)
+            {
+              log.warn(String.format("Failed to push [%,d] segments.", segmentsToPush.size()), e);
+            }
+          }
+      );
+      appenderator.close();
+    }
+
+
+    @Override
+    public void write(NullWritable key, DruidWritable value) throws IOException
+    {
+      InputRow inputRow = new MapBasedInputRow(
+          Longs.tryParse((String) value.getValue()
+                                       .get(DruidTable.DEFAULT_TIMESTAMP_COLUMN)),
+          dataSchema.getParser()
+                    .getParseSpec()
+                    .getDimensionsSpec()
+                    .getDimensionNames(),
+          value.getValue()
+      );
+
+      try {
+
+        appenderator.add(getSegmentIdentifier(value), inputRow, null);
+      }
+      catch (SegmentNotWritableException e) {
+        throw new IOException(e);
+      }
+    }
+
+    @Override
+    public void close(Reporter reporter) throws IOException
+    {
+      appenderator.close();
+    }
+  }
+
+  @Override
+  public FileSinkOperator.RecordWriter getHiveRecordWriter(
+      JobConf jc,
+      Path finalOutPath,
+      Class<? extends Writable> valueClass,
+      boolean isCompressed,
+      Properties tableProperties,
+      Progressable progress
+  ) throws IOException
+  {
+    final String segmentGranularity = tableProperties.getProperty(Constants.DRUID_SEGMENT_GRANULARITY);
+    final String dataSource = tableProperties.getProperty(Constants.DRUID_DATA_SOURCE);
+    final InputRowParser inputRowParser = new MapInputRowParser(new TimeAndDimsParseSpec(
+        new TimestampSpec(DruidTable.DEFAULT_TIMESTAMP_COLUMN, "auto", null),
+        new DimensionsSpec(null, null, null)
+    ));
+    final AggregatorFactory[] aggregatorFactories = DruidStorageHandlerUtils.JSON_MAPPER.readValue(
+        tableProperties.getProperty(Constants.DRUID_AGGREGATORS),
+        AggregatorFactory[].class
+    );
+    final GranularitySpec granularitySpec = new UniformGranularitySpec(
+        Granularity.valueOf(segmentGranularity),
+        null,
+        null
+    );
+    Map<String, Object> inputParser = DruidStorageHandlerUtils.JSON_MAPPER.convertValue(inputRowParser, Map.class);
+    final DataSchema dataSchema = new DataSchema(
+        dataSource,
+        inputParser,
+        aggregatorFactories,
+        granularitySpec,
+        DruidStorageHandlerUtils.JSON_MAPPER
+    );
+
+    // this can be initialized from the hive conf
+    RealtimeTuningConfig realtimeTuningConfig = new RealtimeTuningConfig(
```

Comment:
i am having another PR that will strip bunch of this fields. i know it is confusing but this class can be initilized 100% from hive conf